# Lab 1 Machine Learning

### Authors:
Alfredo Ibias Martínez  
Miguel Benito Parejo

The aim of this Lab is to preprocess and audit a given dataset. In order to do so, we analyse the data to prepare it for future uses on machine learning tasks.  
The given dataset has 32561 entries with the following variables:  
- age
- workclass
- fnlwgt
- education
- education-num
- marital-status
- ocupation
- relationship
- race
- sex
- capital-gain
- capital-loss
- hours-per-week
- native-country

And they are classified in the following two classes:
- <=50K
- \>50K

First of all, we need to import the libraries we will use along the lab.

In [8]:
#Libraries

#import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import sklearn as skl
from sklearn import preprocessing
import matplotlib.pyplot as plt

Now, we can upload our dataset.

In [9]:
# Read Dataset

dataset = pd.read_csv('./Adult_Income.txt',delimiter=',',header=0)
#dataset

# Header
header = []
for row in dataset:
    header.append(row)
#header

Once we have upload our dataset, we face our first problem. It was what to do with unknown data.  
As our database is large, we first decided to check how many entries lack some data. Therefore, we delete these entries from our dataset and check how many entries we still have.

First, lets see what we have.

In [10]:
print(dataset.describe())

                age        fnlwgt   education-num   capital-gain  \
count  32561.000000  3.256100e+04    32561.000000   32561.000000   
mean      38.581647  1.897784e+05       10.080679    1077.648844   
std       13.640433  1.055500e+05        2.572720    7385.292085   
min       17.000000  1.228500e+04        1.000000       0.000000   
25%       28.000000  1.178270e+05        9.000000       0.000000   
50%       37.000000  1.783560e+05       10.000000       0.000000   
75%       48.000000  2.370510e+05       12.000000       0.000000   
max       90.000000  1.484705e+06       16.000000   99999.000000   

        capital-loss   hours-per-week         class  
count   32561.000000     32561.000000  32561.000000  
mean       87.303830        40.437456      0.240810  
std       402.960219        12.347429      0.427581  
min         0.000000         1.000000      0.000000  
25%         0.000000        40.000000      0.000000  
50%         0.000000        40.000000      0.000000  
75%      

In [11]:
pos = []
for column in dataset.columns:
    count = 0
    for elem in dataset[column]:
        if elem == " ?":
            dataset = dataset.drop(count)
            pos.append(count)
        count += 1
        while pos.__contains__(count):
            count += 1

In [6]:
print(dataset.describe())

                age        fnlwgt   education-num   capital-gain  \
count  30162.000000  3.016200e+04    30162.000000   30162.000000   
mean      38.437902  1.897938e+05       10.121312    1092.007858   
std       13.134665  1.056530e+05        2.549995    7406.346497   
min       17.000000  1.376900e+04        1.000000       0.000000   
25%       28.000000  1.176272e+05        9.000000       0.000000   
50%       37.000000  1.784250e+05       10.000000       0.000000   
75%       47.000000  2.376285e+05       13.000000       0.000000   
max       90.000000  1.484705e+06       16.000000   99999.000000   

        capital-loss   hours-per-week         class  
count   30162.000000     30162.000000  30162.000000  
mean       88.372489        40.931238      0.248922  
std       404.298370        11.979984      0.432396  
min         0.000000         1.000000      0.000000  
25%         0.000000        40.000000      0.000000  
50%         0.000000        40.000000      0.000000  
75%      

As we can see, before deleting any entry we had 32561 entries, and now we left 30162 entries.

In [14]:
initial = 32561
final = 30162
deleted = initial-final
deleted

2399

In [13]:
percentage = deleted/initial
percentage

0.07367709836921471

Therefore, we have deleted 2399 entries, that are only a 7.37% of the total entries. Thus, we can assume that loss.